tests/catalog/test_rest.py

```diff
@@ -1226,3 +1226,25 @@ def test_catalog_from_parameters_empty_env(rest_mock: Mocker) -> None:
 
     catalog = cast(RestCatalog, load_catalog("production", uri="https://other-service.io/api"))
     assert catalog.uri == "https://other-service.io/api"
+
+
+def test_commit_table_fail(rest_mock: Mocker, example_table_metadata_v2: Dict[str, Any]) -> None:
```

Comment:
nit: I find the name of this test confusing as it sounds like we are expecting the table commit to fail, but it succeeds with a 200 status code within the test